pkg/ipamd/ipamd.go

```diff
@@ -783,7 +835,36 @@ func (c *IPAMContext) tryAssignIPs() (increasedPool bool, err error) {
 			ipamdErrInc("increaseIPPoolGetENIaddressesFailed")
 			return true, errors.Wrap(err, "failed to get ENI IP addresses during IP allocation")
 		}
-		c.addENIaddressesToDataStore(ec2Addrs, eni.ID)
+
+		c.addENIsecondaryIPsToDataStore(ec2Addrs, eni.ID)
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *IPAMContext) tryAssignPrefixes() (increasedPool bool, err error) {
+	toAllocate := c.getPrefixesNeeded()
```

Comment:
nit: it feels to me we should have a common function that tells how many prefix are needed and how many prefixes are unneeded.

Reply:
As discussed on call, will add it to post GA AI.